lib/commands/release/index.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * @license
+ * Copyright 2016-2020 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { flags } from '@oclif/command';
+import Command from '../../command';
+import * as cf from '../../utils/common-flags';
+import { getBalenaSdk, getVisuals, stripIndent } from '../../utils/lazy';
+import type * as BalenaSdk from 'balena-sdk';
+import jsyaml = require('js-yaml');
+import { getRelease } from '../../utils/release-utils';
+
+interface FlagsDef {
+	help: void;
+	composition?: boolean;
+}
+
+interface ArgsDef {
+	commitOrId?: string;
+}
+
+export default class ReleaseCmd extends Command {
+	public static description = stripIndent`
+		Get info for a release.
+`;
+	public static examples = ['$ balena release commit', '$ balena release id'];
+
+	public static usage = 'release <commitOrId>';
+
+	public static flags: flags.Input<FlagsDef> = {
+		help: cf.help,
+		composition: flags.boolean({
+			default: false,
+			char: 'c',
+			description: 'Return the release composition',
+		}),
+	};
+
+	public static args = [
+		{
+			name: 'commitOrId',
+			description: 'the commit or ID of the release to get information',
+		},
+	];
+
+	public static authenticated = true;
+
+	public async run() {
+		const { args: params, flags: options } = this.parse<FlagsDef, ArgsDef>(
+			ReleaseCmd,
+		);
+		if (!params.commitOrId) {
+			const { ExpectedError } = await import('../../errors');
+			throw new ExpectedError('You must specify a release commit');
+		}
+
+		const balena = getBalenaSdk();
+		if (options.composition) {
+			await this.showComposition(params.commitOrId, balena);
+		} else {
+			await this.showReleaseInfo(params.commitOrId, balena);
+		}
+	}
+
+	async showComposition(commitOrId: string, balena: BalenaSdk.BalenaSDK) {
+		const release = await getRelease(commitOrId, ['composition'], balena);
+
+		console.log(jsyaml.dump(release.composition));
+	}
+
+	async showReleaseInfo(commitOrId: string, balena: BalenaSdk.BalenaSDK) {
+		const fieldsForGet: Array<keyof BalenaSdk.Release> = [
+			'id',
+			'commit',
+			'created_at',
+			'status',
+			'semver',
+			'is_final',
+			'build_log',
+			'start_timestamp',
+			'end_timestamp',
+		];
+
+		const release = await getRelease(commitOrId, fieldsForGet, balena);
+
+		const tagFields: Array<keyof BalenaSdk.ResourceTagBase> = [
+			'tag_key',
+			'value',
+		];
+
+		const tags = await balena.models.release.tags.getAllByRelease(release.id, {
+			$select: tagFields,
+		});
```

Comment:
Pine allows you to combine those two requests into a single one, so that we are kinder with our API:
```js
await balena.models.release.get(commitOrId, {
	$select: fieldsForGet,
	$expand: {
		release_tag: {
			$select: tagFields,
		}
	}
});
```
One extra reason for using `tryAsInteger` or `tryIdOrStringKeyGet` since they will make it easier to do so.

Reply:
Perfect.  I was wondering how to do this better.